compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/AnvilModuleDescriptor.kt

```diff
@@ -0,0 +1,31 @@
+package com.squareup.anvil.compiler.internal
+
+import com.squareup.anvil.annotations.ExperimentalAnvilApi
+import org.jetbrains.kotlin.descriptors.ModuleDescriptor
+import org.jetbrains.kotlin.name.ClassId
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.name.Name
+
+@ExperimentalAnvilApi
+public abstract class AnvilModuleDescriptor : ModuleDescriptor {
+  public abstract fun resolveClassIdOrNull(classId: ClassId): FqName?
+}
+
+public fun ModuleDescriptor.resolveFqNameOrNull(
+  fqName: FqName
+): FqName? = (this as AnvilModuleDescriptor).resolveClassIdOrNull(fqName.classIdBestGuess())
```

Comment:
This cast is the scary thing. But the code lives in the utils module, which gives no guarantees about compatibility and API stability. I suggest to read #283 and #282 to understand why this is needed.